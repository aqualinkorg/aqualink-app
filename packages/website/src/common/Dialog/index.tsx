import React from "react";
import { Link } from "react-router-dom";
import {
  withStyles,
  WithStyles,
  createStyles,
  Theme,
  Dialog as MuiDialog,
  DialogTitle,
  DialogActions,
  DialogContent,
  Typography,
  ButtonProps,
  Button,
} from "@material-ui/core";

const Dialog = ({
  open,
  header,
  content,
  actions,
  onClose,
  classes,
}: DialogProps) => {
  return (
    <MuiDialog fullWidth onClose={onClose} open={open}>
      <DialogTitle className={classes.dialogTitle}>
        <Typography>{header}</Typography>
      </DialogTitle>
      <DialogContent>{content}</DialogContent>
      <DialogActions>
        {actions &&
          actions.map((action) => {
            if (action.link) {
              return (
                <Link
                  key={action.text}
                  style={{ color: "inherit", textDecoration: "none" }}
                  to={action.link}
                >
                  <Button
                    key={action.text}
                    color={action.color}
                    variant={action.variant}
                    size={action.size}
                    disabled={action.disabled}
                    onClick={action.action}
                  >
                    {action.text}
                  </Button>
                </Link>
              );
            }
            return (
              <Button
                key={action.text}
                color={action.color}
                variant={action.variant}
                size={action.size}
                disabled={action.disabled}
                onClick={action.action}
              >
                {action.text}
              </Button>
            );
          })}
      </DialogActions>
    </MuiDialog>
  );
};

const styles = (theme: Theme) =>
  createStyles({
    dialogTitle: {
      backgroundColor: theme.palette.primary.main,
      overflowWrap: "break-word",
    },
  });

export interface Action {
  size: ButtonProps["size"];
  variant: ButtonProps["variant"];
  color: ButtonProps["color"];
  text: string;
  link?: string;
  disabled?: boolean;
  action: (...args: any[]) => void;
}

interface DialogIncomingProps {
  open: boolean;
  header?: string;
  content?: JSX.Element | null;
  actions?: Action[];
  onClose?: (...args: any[]) => void;
}

Dialog.defaultProps = {
  header: "",
  content: null,
  actions: [],
  onClose: () => {},
};

type DialogProps = DialogIncomingProps & WithStyles<typeof styles>;

export default withStyles(styles)(Dialog);
