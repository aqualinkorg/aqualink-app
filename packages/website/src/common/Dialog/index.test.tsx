import React from "react";
import { render } from "@testing-library/react";
import Dialog from ".";

test("renders as expected", () => {
  const { container } = render(
    <Dialog
      open
      onClose={() => {}}
      header="Random Header"
      content={<div>Some random content</div>}
      actions={[]}
    />
  );
  expect(container).toMatchSnapshot();
});
