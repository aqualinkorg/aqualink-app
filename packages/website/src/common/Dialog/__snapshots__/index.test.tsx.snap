// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders as expected 1`] = `
<div>
  <mock-dialog
    fullwidth="true"
    open="true"
  >
    <mock-dialogtitle
      classname="Dialog-dialogTitle-1"
    >
      <mock-typography>
        Random Header
      </mock-typography>
    </mock-dialogtitle>
    <mock-dialogcontent>
      <div>
        Some random content
      </div>
    </mock-dialogcontent>
    <mock-dialogactions />
  </mock-dialog>
</div>
`;
